### PR TITLE
added typescript

### PR DESCRIPTION
**Pull request by mip92**

### CODE REVIEW

File: src/controller/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            }))
+            let comparePassword: boolean = bcrypt.compareSync(password, user.password)
+            if (!comparePassword) return next(ApiError.ExpectationFailed({
+                value: email,
+                msg: "User is not found or password is wrong",
+                param: "email",
+                location: "body"
+            }))
+            const token = tokenService.generateJwt(user.id, user.email, user.role)
+            return res.status(200).json({token, name: "name" in user ? user.name : "admin"})
```
Comment (IvashchenkoCW) [inline src/controller/auth.controller.ts:44]: you can use 'user.name ?? "admin"'

File: src/controller/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        } catch (e) {
+            console.log(e)
```
Comment (IvashchenkoCW) [inline src/controller/auth.controller.ts:46]: I guess you should make a response error

File: src/controller/auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            if (firstPassword !== secondPassword) {
+                return next(ApiError.BadRequest('Passwords do not match'))
+            }
+            if (!isRulesChecked) {
+                return next(ApiError.BadRequest('Use of service rules is not confirmed'))
+            }
+            if (!isMaster) await userController.registration(req, res, next)
+            else await masterController.registration(req, res, next)
+        } catch (e) {
+            console.log(e)
```
Comment (IvashchenkoCW) [inline src/controller/auth.controller.ts:62]: I guess you should make a response error

File: src/controller/auth.controller.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                return res.redirect(`${process.env.CLIENT_URL}`);
+            }
+            const master: MasterModel = await Master.findOne({where: {activationLink}})
+            if (master){
+                await master.update({isActivated: true})
+                return res.redirect(`${process.env.CLIENT_URL}`);
+            }
+            return next(ApiError.BadRequest('Incorrect activation link'))
+        } catch (e) {
+            console.log(e)
```
Comment (IvashchenkoCW) [inline src/controller/auth.controller.ts:81]: I guess you should make a response error

File: src/controller/auth.controller.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                return res.redirect(process.env.CLIENT_URL + '/login');
+            }
+            const master: MasterModel = await Master.findOne({where: {activationLink}})
+            if (master){
+                await master.update({isActivated: true})
+                return res.redirect(process.env.CLIENT_URL + '/login');
+            }
+            return next(ApiError.BadRequest('Incorrect activation link'))
+        } catch (e) {
+            console.log(e)
```
Comment (IvashchenkoCW) [inline src/controller/auth.controller.ts:100]: I guess you should make a response error

File: src/controller/order.controller.ts [hunk 6]
```diff
@@ -1,59 +1,66 @@
...
-                    activationLink
+                    activationLink,
                 })
-                const master = await Master.findOne({where: {id: masterId}})
-                const city = await City.findOne({where: {id: cityId}})
-                let arr = []
-                for (let i = 0; i < clockSize; i++) {
+                const master: MasterModel = await Master.findOne({where: {id: masterId}})
+                const city: CityModel = await City.findOne({where: {id: cityId}})
+                let arr = [] as number[]
```
Comment (IvashchenkoCW) [inline src/controller/order.controller.ts:33]: let arr: number[] = []
Make name of array with sense

File: src/controller/order.controller.ts [hunk 7]
```diff
@@ -162,15 +170,16 @@ class OrderController {
...
 
-    async getAllOrders(req, res, next) {
+    async getAllOrders(req: CustomRequest<null, null, GetAllOrders, null>, res: Response, next: NextFunction) {
         try {
             let {limit, offset, masterId, userId} = req.query
-            if (limit > 50) limit = 50
-            if (!offset) offset = 0
+            if (limit && +limit > 50) limit = '50'
+            if (!offset) offset = '0'
             let where
```
Comment (IvashchenkoCW) [inline src/controller/order.controller.ts:178]: const where = {}
if(userId) where.userId = userId

File: src/controller/order.controller.ts [hunk 8]
```diff
@@ -162,15 +170,16 @@ class OrderController {
...
-            if (userId) where = {'userId': userId}
-            else if (masterId) where = {'masterId': masterId}
-            const options = {
+            if (userId) where = {'userId': +userId}
+            else if (masterId) where = {'masterId': +masterId}
+
+            const options: any = {
```
Comment (IvashchenkoCW) [inline src/controller/order.controller.ts:182]: You should import from sequelize interface of payload of findAndCountAll methode

File: src/controller/picture.controller.ts [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+class PictureController {
+    static createOnePicture(file: MyFile, next: NextFunction) {
+        const name: string = file.name
+        const fileExtension: string | undefined = name.split('.').pop()
+        const arr: string[] = ['jpeg', 'JPEG', 'jpg', 'JPG', 'png', 'PNG'] //'BMP', 'bmp', 'GIF', 'gif', 'ico', 'ICO'
+        let isOk: null | string = null
+        for (let i = 0; i < arr.length; i++) {
+            if (fileExtension == arr[i]) isOk = "ok"
+        }
+        if (isOk == null) return next(ApiError.BadRequest(`Файла ${name} не является картинкой`))
```
Comment (IvashchenkoCW) [inline src/controller/picture.controller.ts:64]: English?

File: src/controller/picture.controller.ts [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+interface OrderPictureModelWithPicture extends OrderPictureModel {
+    picture: Picture
+}
+
+class PictureController {
+    static createOnePicture(file: MyFile, next: NextFunction) {
+        const name: string = file.name
+        const fileExtension: string | undefined = name.split('.').pop()
+        const arr: string[] = ['jpeg', 'JPEG', 'jpg', 'JPG', 'png', 'PNG'] //'BMP', 'bmp', 'GIF', 'gif', 'ico', 'ICO'
```
Comment (IvashchenkoCW) [inline src/controller/picture.controller.ts:59]: What is it 'arr'? Make some names with sense. For example 'availableFileTypes' etc.

File: src/controller/picture.controller.ts [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+        let isOk: null | string = null
+        for (let i = 0; i < arr.length; i++) {
```
Comment (IvashchenkoCW) [inline src/controller/picture.controller.ts:61]: FOOOORR?

File: src/controller/picture.controller.ts [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+            )
+        } catch (e) {
+            console.log(e)
+            next(ApiError.Internal(`server error`))
+        }
+    }
+
+    async deletePictures(req: CustomRequest<DeletePicturesBody, CreatePicturesParams, null, null>, res: Response, next: NextFunction) {
+        const {orderId} = req.params
+        const arr = /*JSON.parse(*/req.body.picturesId
```
Comment (IvashchenkoCW) [inline src/controller/picture.controller.ts:159]: arr?

File: src/controller/rating.controller.ts [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+    async getRatingByMaster(req: CustomRequest<null, GetRatingByMasterParams, null, null>, res: Response, next: NextFunction) {
+        try {
+            const masterId = req.params.masterId
+            const ratings: RatingModel[] = await Rating.findAll({
+                    where: {masterId: +masterId},
+                }
+            )
+            if (!ratings) return next(ApiError.BadRequest("Ratings not found"))
+            let arr=[] as number[]
```
Comment (IvashchenkoCW) [inline src/controller/rating.controller.ts:53]: let arr: number[] = []

File: src/controller/auth.controller.ts (hunk 1 above)
Comment (mip92) [reply src/controller/auth.controller.ts:43]: Не получается так . Ошибку выдает: "Property 'name' does not exist on type 'AdminModel | UserModel | MasterModel'.   Property 'name' does not exist on type 'AdminModel'."

File: src/controller/city.controller.ts [hunk 14]
```diff
@@ -43,6 +42,7 @@ class CityController {
                     ],
                 }
             )
+            console.log(cities)
```
Comment (IvashchenkoCW) [inline src/controller/city.controller.ts:45]: Remove

File: src/controller/picture.controller.ts [hunk 15]
```diff
@@ -56,39 +55,36 @@ class PictureController {
...
-        let isOk: null | string = null
-        for (let i = 0; i < arr.length; i++) {
-            if (fileExtension == arr[i]) isOk = "ok"
+        const allowedTypes: string[] = ['jpeg', 'JPEG', 'jpg', 'JPG', 'png', 'PNG'] //'BMP', 'bmp', 'GIF', 'gif', 'ico', 'ICO'
+        if (!allowedTypes.some(fileType => fileType === fileExtension)){
+            return next(ApiError.BadRequest(`File with name: ${name} is not a picture`))
         }
-        if (isOk == null) return next(ApiError.BadRequest(`Файла ${name} не является картинкой`))
-        if (file.size > 1048576) return next(ApiError.BadRequest(`Размер файла ${name} больше 1 Мб`))
+        if (file.size > 1048576) return next(ApiError.BadRequest(`File with name: ${name} is larger than 1 MB`))
```
Comment (IvashchenkoCW) [inline src/controller/picture.controller.ts:62]: Declare const variable in the start of file like: const MAX_FILE_SIZE

File: src/routes/payPal.router.ts [hunk 16]
```diff
@@ -0,0 +1,17 @@
+import {NextFunction} from "express";
```
Comment (IvashchenkoCW) [inline src/routes/payPal.router.ts:1]: Why do you add this file to this PR?

**Pull request merged**